clang/lib/Sema/Sema.cpp

```diff
@@ -1860,6 +1860,10 @@ class DeferredDiagnosticsEmitter
   void checkFunc(SourceLocation Loc, FunctionDecl *FD) {
     auto &Done = DoneMap[InOMPDeviceContext > 0 ? 1 : 0];
     FunctionDecl *Caller = UsePath.empty() ? nullptr : UsePath.back();
+
+    if (!Caller)
```

Comment:
This should be restricted to SYCL contexts - we get here for OpenMP and CUDA as well.

Reply:
Okay, I will try to restrict it.